lib/proxy/pluggable.rb

```diff
@@ -1,131 +1,137 @@
 module Proxy::Pluggable
-  def self.included(base)
-    base.send(:include, InstanceMethods)
-    base.send(:extend, ClassMethods)
+  attr_reader :plugin_name, :version, :after_activation_blk, :class_loader
+
+  # Methods below define DSL for defining plugins
+  def after_activation(&blk)
```

Comment:
Is this method meant to be deprecated?  There are references in the loading code that it's "legacy".